--- pcl/pl/pjparse.c.old	2017-07-21 21:00:16.554676701 +0200
+++ pcl/pl/pjparse.c	2017-07-21 21:01:37.505672224 +0200
@@ -124,7 +124,8 @@
    source table. */
 static pjl_fontsource_t pjl_fontsource_table[] = {
     {"I",
-     "%rom%ttfonts/;urwfonts/;pcl/urwfonts/;ghostpdl/pcl/urwfonts/;/windows/fonts/;", ""},
+//     "%rom%ttfonts/;urwfonts/;pcl/urwfonts/;ghostpdl/pcl/urwfonts/;/windows/fonts/;", ""},
+     "%rom%ttfonts/;urwfonts/;pcl/urwfonts/;ghostpdl/pcl/urwfonts/;/usr/share/fonts/pclfonts/;", ""},
     {"C", "CART0/", ""},
     {"C1", "CART1/", ""},
     {"C2", "CART2/", ""},
