diff -Nru coolwanglu-pdf2htmlEX-d286233/src/util/unicode.cc coolwanglu-pdf2htmlEX-d286233-mod/src/util/unicode.cc
--- coolwanglu-pdf2htmlEX-d286233/src/util/unicode.cc	2014-01-19 12:39:05.000000000 +0100
+++ coolwanglu-pdf2htmlEX-d286233-mod/src/util/unicode.cc	2014-07-03 13:14:24.537203402 +0200
@@ -8,6 +8,7 @@
 #include <iostream>
 
 #include <GlobalParams.h>
+#include <cpp/poppler-version.h>
 
 #include "unicode.h"
 
@@ -118,7 +119,11 @@
         // may be untranslated ligature
         if(cname)
         {
+#if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR > 24
+            Unicode ou = globalParams->mapNameToUnicodeText(cname);
+#else
             Unicode ou = globalParams->mapNameToUnicode(cname);
+#endif
 
             if(isLegalUnicode(ou))
                 return ou;
