diff -Nru numpy-1.11.2/numpy/core/src/multiarray/numpyos.c numpy-1.11.2-fixed/numpy/core/src/multiarray/numpyos.c
--- numpy-1.11.2/numpy/core/src/multiarray/numpyos.c	2016-10-04 01:42:24.000000000 +0200
+++ numpy-1.11.2-fixed/numpy/core/src/multiarray/numpyos.c	2018-08-27 14:22:24.071170923 +0200
@@ -15,7 +15,7 @@
 
 #ifdef HAVE_STRTOLD_L
 #include <stdlib.h>
-#include <xlocale.h>
+#include <locale.h>
 #endif
 
 
