--- src/lib/net/CSocketMultiplexer.cpp.orig	2011-08-11 17:08:31.000000000 -0500
+++ src/lib/net/CSocketMultiplexer.cpp	2011-08-11 17:11:09.000000000 -0500
@@ -255,6 +255,7 @@
 		for (CSocketJobMap::iterator i = m_socketJobMap.begin();
 							i != m_socketJobMap.end();) {
 			if (*(i->second) == NULL) {
+				m_socketJobs.erase(i->second);
 				m_socketJobMap.erase(i++);
 				m_update = true;
 			}
